extend/flytekit-xgboost/flytekitplugins/xgboost/task.py

```diff
@@ -129,25 +167,19 @@ def __init__(
         )
 
     # Train method
-    def train(
-        self, dtrain: xgboost.DMatrix, dvalid: xgboost.DMatrix, **kwargs
-    ) -> Tuple[str, Dict[str, Dict[str, List[float]]]]:
+    def train(self, dtrain: xgboost.DMatrix, dvalid: xgboost.DMatrix, params: XGBoostParameters) -> Tuple[str, Dict[str, Dict[str, List[float]]]]:
         evals_result = {}
         # if validation data is provided, then populate evals and evals_result
         if dvalid:
-            booster_model = xgboost.train(
-                params=asdict(self._config.hyper_parameters),
-                dtrain=dtrain,
-                **asdict(self._config.model_parameters),
-                evals=[(dvalid, "validation")],
-                evals_result=evals_result,
-            )
-        else:
-            booster_model = xgboost.train(
-                params=asdict(self._config.hyper_parameters),
-                dtrain=dtrain,
-                **asdict(self._config.model_parameters),
-            )
+            validation = [(dvalid, "validation")]
```

Comment:
We have to initialize validation first because this doesn't work if `dvalid` evaluates to False.